src/markdown_exec/_internal/mkdocs_plugin.py

```diff
@@ -125,15 +154,24 @@ def on_post_build(self, *, config: MkDocsConfig) -> None:  # noqa: ARG002
         """Reset the plugin state."""
         MarkdownConverter.counter = 0
         markdown_config.reset()
-        if self.mkdocs_config_dir is None:
-            os.environ.pop("MKDOCS_CONFIG_DIR", None)
-        else:
-            os.environ["MKDOCS_CONFIG_DIR"] = self.mkdocs_config_dir
 
-    def _add_asset(self, config: MkDocsConfig, asset_file: str, asset_type: str) -> None:
+        # Restore original environment variables
+        for var, value in self.original_env_vars.items():
+            if value is None:
+                os.environ.pop(var, None)
+            else:
+                os.environ[var] = value
+
+    def _add_asset(
+        self, config: MkDocsConfig, asset_file: str, asset_type: str
+    ) -> None:
         asset_filename = f"assets/_markdown_exec_{asset_file}"
-        asset_content = Path(__file__).parent.parent.joinpath("assets", asset_file).read_text()
-        write_file(asset_content.encode("utf-8"), os.path.join(config.site_dir, asset_filename))
+        asset_content = (
+            Path(__file__).parent.parent.joinpath("assets", asset_file).read_text()
+        )
+        write_file(
+            asset_content.encode("utf-8"), os.path.join(config.site_dir, asset_filename)
+        )
```

Comment:
Please do not reformat code blocks that are unrelated to the core changes in this PR